### PR TITLE
DDPB-3947: Behat rewrite - report management

### PR DESCRIPTION
**Pull request by acsauk**
## Purpose
Covers the report management section of the app on the admin side.

Fixes DDPB-3947

## Approach
I took the opportunity to:

* Drop out a bunch of properties from `FixtureHelper` that was a hangover from when behat was separate to API. We were just setting them and then passing them to a function on the next line down so seemed pointless.
* Make some of the fixture users more explicit in their naming for report types
* Use DateTime objects for UserDetails properties rather than strings to expand their usage in tests
* Fixed the pesky en dash used in the report period on client overview page

### CODE REVIEW

File: api/tests/Behat/bootstrap/v2/ClientManagement/ClientManagementTrait.php [hunk 1]
```diff
@@ -16,7 +16,7 @@ trait ClientManagementTrait
      */
     public function iSearchForExistingClientByFirstName()
     {
-        $user = is_null($this->interactingWithUserDetails) ? $this->profAdminDeputyNotStartedDetails : $this->interactingWithUserDetails;
+        $user = is_null($this->interactingWithUserDetails) ? $this->profAdminDeputyHealthWelfareNotStartedDetails : $this->interactingWithUserDetails;
```
Comment (jamesrwarren) [inline api/tests/Behat/bootstrap/v2/ClientManagement/ClientManagementTrait.php:19]: Nice. Kept meaning to do this! :)

**Review by coderste — DISMISSED** [review]

File: api/tests/Behat/bootstrap/v2/ReportManagement/ReportManagementTrait.php [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+     * @When I change the report type to :reportType
+     */
+    public function iChangeReportTypeTo(string $reportType)
+    {
+        $this->iAmOnAdminManageReportPage();
+        $this->assertInteractingWithUserIsSet();
+
+        $isLay = is_null($this->interactingWithUserDetails->getOrganisationName);
+
+        switch ($reportType) {
```
Comment (jamesrwarren) [inline api/tests/Behat/bootstrap/v2/ReportManagement/ReportManagementTrait.php:70]: Bear in mind. You're doing prod and lay here but not PA. Pa report types are different. This comment might be pointless.. haven't got to how you're using this yet ;)

File: api/tests/Behat/features-v2/report-management/report-management.feature [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    Scenario: An admin user changes report type and due date for an un-submitted report
+        Given an admin user accesses the admin app
+        And a Public Authority Deputy has submitted a Combined High Assets report
+        When I visit the admin client details page associated with the deputy I'm interacting with
+        And I manage the deputies 'submitted' report
+        And I confirm all report sections are incomplete
+        And I submit the new report details
+        And I manage the deputies 'un-submitted' report
+        And I change the report due date to '5' weeks from now
+        And I change the report type to 'Health and welfare'
```
Comment (jamesrwarren) [inline api/tests/Behat/features-v2/report-management/report-management.feature:40]: Yeah so you ought to use the PA types as per above comment I believe :)

File: client/templates/Admin/Client/Client/details.html.twig [hunk 4]
```diff
@@ -93,16 +93,16 @@
...
-                {{ 'phone' | trans({}, 'common') }}: {{ namedDeputy.phoneMain }}<br>
+                {{ 'phone' | trans({}, 'common') }}: {{ namedDeputy.phoneMain }}<br />
             {% endif %}
             {% if namedDeputy.email %}
-                {{ 'email' | trans({}, 'common') }}: {{ namedDeputy.email }}<br>
+                {{ 'email' | trans({}, 'common') }}: {{ namedDeputy.email }}<br />
             {% endif %}
             {% if namedDeputy.email2 is defined and namedDeputy.email2 %}
-                {{ 'alternativeEmail' | trans({}, 'common') }}: {{ namedDeputy.email2 }}<br>
+                {{ 'alternativeEmail' | trans({}, 'common') }}: {{ namedDeputy.email2 }}<br />
```
Comment (jamesrwarren) [inline client/templates/Admin/Client/Client/details.html.twig:102]: Nice ;) xpath you picky bar steward!

File: api/tests/Behat/features-v2/report-management/report-management.feature (hunk 3 above)
Comment (acsauk) [reply api/tests/Behat/features-v2/report-management/report-management.feature:40]: Ah good point - I always forget that PA have a slightly different report layout.

**Review by jamesrwarren — APPROVED** [review]

**Pull request merged**